unit test compile error Msvc  Release

### Search before asking

- [ ] I searched the [issues](https://github.com/alibaba/yalantinglibs/issues) and found no similar issues.
 
`MSVC+Release+ninja/MSBuild/NMake` have the same error
error C4737: 无法执行所需尾调用。性能可能会降级。

> WINDOW11 + VS 2022  
> HOST : WINDOW X64
> TARGET: WINDOW X64
> MSVC: CXX compiler identification is MSVC 19.34.31935.0
> SDK: Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22621.


### What happened + What you expected to happen

![image](https://user-images.githubusercontent.com/70421213/206052022-33d60b8c-a602-4b01-8344-5efe72937c67.png)

--- a/cmake/platform.cmake
+++ b/cmake/platform.cmake
@@ -10,3 +10,5 @@ endif()
 # --------------------- Msvc
 # Resolves C1128 complained by MSVC: number of sections exceeded object file format limit: compile with /bigobj.
 add_compile_options("$<$<CXX_COMPILER_ID:MSVC>:/bigobj>")
+# Resolves C4737 complained by MSVC: C4737: Unable to perform required tail call. Performance may be degraded.
+add_compile_options("$<$<CXX_COMPILER_ID:MSVC>:/EHa>")

